### PR TITLE
feat: add ListJobs map for greenhouse

### PR DESCRIPTION
**Pull request by tmladek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/recruitment/list-jobs/maps/greenhouse.suma [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  limit = input.limit || 100
+  
+  http GET "/v1/jobs" {
+    security "apiKeyBasicAuth"
+  
+    request {
+      query {
+        page = page
+        per_page = limit
+        status = JOB_STATES_S2G[input.state]
```
Comment (jnv) [inline grid/recruitment/list-jobs/maps/greenhouse.suma:29]: issue: Since state is not mandatory, I may end up passing `JOB_STATES_S2G[undefined]` here.

**Review by jnv — CHANGES_REQUESTED** [review]
Looks good, I see one critical issue.

File: grid/recruitment/list-jobs/maps/greenhouse.suma (hunk 1 above)
Comment (tmladek) [reply grid/recruitment/list-jobs/maps/greenhouse.suma:29]: In which case it evaluates to `undefined` and isn't passed in the `query` at all, right?

Comment (jnv) [reply grid/recruitment/list-jobs/maps/greenhouse.suma:29]: Right, it should, I just don't like that it relies on implicit behavior. Feel free to ignore this suggestion then.

```suggestion
        status = input.state && JOB_STATES_S2G[input.state]
```

(even better would be to assign state above the http call)

**Review by jnv — APPROVED** [review]

**Pull request merged**